src/objects/zcl_abapgit_object_prog.clas.abap

```diff
@@ -145,37 +145,41 @@ CLASS ZCL_ABAPGIT_OBJECT_PROG IMPLEMENTATION.
 
     lv_program_name = ms_item-obj_name.
 
-    lt_source = mo_files->read_abap( ).
+    CASE iv_step.
+      WHEN zif_abapgit_object=>gc_step_id-abap_before_ddic.
```

Comment:
whats the reason for importing PROG before DDIC?

Reply:
@larshp from what I know this step is crucial to this change. If a DDIC structure has a field "type ref to some abap object" there is a need to import this abap object before activating DDIC.

Reply:
Yes, that's the point.
To adress the issue that some DDIC can not be imported when abap coding is missing (search help with function module or tabletype with classes).

So it ist requird that the abap is in the system beforde the DDIC.
But therefore the abap has to be activ eith syntax error cause the DDIC is not in the system becuse it can ony be imported when the abap is in the systen.

> The deserialize order was changed ddic is now imported after the first abap step
For that reason it is required to activate the abap objects even with syntax error (not possible for ddic elements).

For some abap elemens a special treatment is required (cause... :)):
* Remote function modules whant to check the parameters if RFC is paussible. Therefore RFC is active only after DDIC was imported
* Dynpros need the DDIC active in the system

Reply:
but these are PROG objects? I cannot do `TYPE REF TO zprogram`, am I missing something?

Reply:
No it is correct that not all abap objects (like e.g. PROG) have to be imported before DDIC. I simply want to keep the steps kind of clean and don't want to mix the objects.
So I focus on the goal to import "as much as possible" that can be forced into the system.

It would be possible to import some object early then import DDIC then import the rest of the objects ant the import the late objects.
But I think this would make it more complex.